spark/src/main/java/org/apache/iceberg/spark/source/RowDataReader.java

```diff
@@ -140,6 +142,9 @@
       builder.withNameMapping(NameMappingParser.fromJson(nameMapping));
     }
 
+    // TODO: propagate properties for Avro
+    // properties.forEach(builder::set);
```

Comment:
Maybe we should have an Precondition til this is implemented throwing an exception if a property is passed

Reply:
As it stands today, all table props are passed here so this precondition will always fail. We may want to consider setting only those table props that start with `read.parquet` for Parquet or `read.avro` for Avro. Then a precondition here would work as we ensure props only contain relevant configs.

Reply:
We can do that in our builders but that will be extra work.